### DISCUSSION TITLE
correlation order vs layers

### DISCUSSION THREAD

**Discussion by JSLJ23** [ACEsuit/mace#289] [Q&A] [answered]
Hi MACE developers,

I am reading through the [MACE paper](https://arxiv.org/pdf/2206.07697.pdf) but I am also having quite a bit of difficulty understanding the V parameter which is supposed to represent the "correlation order".
My understanding is that this corresponds to the number of message passing operations performed on each node and I don't quite understand how this differs from the number of layers...?
Doesn't the number of layers in a MPNN directly influence the number of message passing steps and hence this "correlation order"?
In figure 2, the caption describes a two layer invariant (L = 0) MACE with increasing body order ν ∈ {1, 2, 3}. How is the body order able to increase independent of the number of layers?

Thank you for your time to address my queries.

**Comment by wcwitt** [comment]
Hi - thanks for the interest. The correlation order \nu controls the expressiveness within a single layer: as \nu increases, the ACE-like basis describing the environment around a single central atom (node) becomes larger and more complex. This information is condensed in messages that are passed between nodes; the parameter L controls the expressiveness of these messages. And the number of layers is a separate parameter.

**Reply by JSLJ23** [reply]
Could you give further elaboration on what the "expressiveness within a single layer" means in terms of the underlying operations that happen on the MPNN?

**Reply by wcwitt** [reply, marked as answer]
Imagine a central atom at the origin and suppose its environment includes N neighbors within some cutoff. You can describe the environment completely with 4N+1 numbers (3N xyz coordinates and N+1 atom types). However, for various reasons, it is not convenient to work in this "basis" and so one of the first steps is to translate this information into a different basis. By "expressiveness within a single layer", I'm referring to the size/expressiveness of this auxiliary basis. For more, I recommend looking at the ACE literature and/or here: https://arxiv.org/pdf/2205.06643.pdf.

**Reply by gabor1** [reply]
I was just going to say: the only way to understand this is to look at the ACE papers, which are high body order (but there is no message passing)

**Reply by JSLJ23** [reply]
Ok will do, thank you!

**Comment by gabor1** [comment]
If you look at equation 7 in this paper: https://doi.org/10.1063/5.0155322 this is where \nu is used, and you can see that it is all still centred on atom i, it's building high correlation order (high body order) representations of the local geometry by multiplying together the atomic basis \nu times. the message passing happens when you iterate equations (2)-(10) multiple times (and we typically only do it 2 times)
